### PR TITLE
LG-4105 codemods

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-4105](https://jira.mongodb.org/browse/LG-4105)
🎟 _Jira ticket:_ [LG-4106](https://jira.mongodb.org/browse/LG-4106)
🎟 _Jira ticket:_ [LG-4107](https://jira.mongodb.org/browse/LG-4107)
🎟 _Jira ticket:_ [LG-4108](https://jira.mongodb.org/browse/LG-4108)

All the reusable transformation utils live in `src/utils/transformations/`. 

To test the migrations, I created example codemod in `src/codemods`.
Each example directory includes: 
- a `transform.ts` file, which is the transformation code that `jscodeshift` will execute for each file passed to it.
- a `tests` directory that has tests for each transformation. Each test has an `input` file and `output` file. The `input` file and `transform.js` is passed to `jscodeshift` and the output from `jscodeshift` is compared against the `output` file. If they match then the test will pass.

To test codemod in jest
`yarn test /tools/codemods`

## 🐞 Current Bugs
- ~When using a glob pattern, `jscodeshift` will add parentheses around pieces of code inside files that don't need to be touched at all.~
- ~`is-git-clean` can't be found error in file size check~
- ~every `dist` folder in every package has changed~
- file size check error

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: tools/migrator/src/utils/tests/transformCheck.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import path from 'path';
+// @ts-expect-error - no prettier types
+import prettier from 'prettier';
+
+async function applyTransform(
+  transform: any,
+  input: FileInfo,
+  options?: { [option: string]: any },
+) {
+  // TODO: don't understand whats happening here
```
Comment (shaneeza) [inline tools/migrator/src/utils/tests/transformCheck.ts:15]: remove because i understand now

File: tools/cli/src/index.ts [hunk 2]
```diff
@@ -166,6 +167,32 @@ cli
   )
   .action(validate);
 
+/** Migrator */
+cli
+  .command('migrator')
+  .description('yada yada')
```
Comment (shaneeza) [inline tools/cli/src/index.ts:173]: update

File: tools/cli/src/index.ts [hunk 3]
```diff
@@ -166,6 +167,32 @@ cli
...
+  .argument(
+    '<migration>',
+    'One of the migrations from: ADD LINK WITH ALL THE MODS',
```
Comment (shaneeza) [inline tools/cli/src/index.ts:176]: Add Link

File: tools/migrator/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+### NPM
+
+```shell
+npm install @lg-tools/migrator
+```
+
+## Usage
+
+//TODO: Update me
```
Comment (shaneeza) [inline tools/migrator/README.md:21]: Update

File: tools/migrator/rollup.config.mjs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import * as path from 'path';
+import * as url from 'url';
+import { glob } from 'glob';
+
+const __dirname = path.dirname(url.fileURLToPath(import.meta.url));
+const pkg = JSON.parse(fs.readFileSync(path.join(__dirname, 'package.json')));
+
+const migratorGlob = glob.sync('./src/migrations/*!(tests)/*.ts');
+
+console.log('🛑🛑🛑🛑', {
```
Comment (shaneeza) [inline tools/migrator/rollup.config.mjs:12]: rm

File: tools/migrator/src/utils/jsx/insertJSXComment/transform.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable no-console */
```
Comment (shaneeza) [inline tools/migrator/src/utils/jsx/insertJSXComment/transform.ts:1]: rm

File: tools/migrator/src/utils/jsx/insertJSXComment/insertJSXComment.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import type { ASTPath, Comment } from 'jscodeshift';
+import type core from 'jscodeshift';
+
+export function insertJSXComment(
```
Comment (shaneeza) [inline tools/migrator/src/utils/jsx/insertJSXComment/insertJSXComment.ts:4]: Add docs

File: tools/cli/src/index.ts (hunk 2 above)
Comment (bruugey) [reply tools/cli/src/index.ts:173]: lol yada yada

File: package.json [hunk 8]
```diff
@@ -29,7 +29,8 @@
     "start": "npx storybook dev -p 9001 --no-version-updates",
     "test": "lg test",
     "unlink": "lg unlink",
-    "validate": "lg validate"
+    "validate": "lg validate",
+    "jscodeshift":"0.15.2"
```
Comment (shaneeza) [inline package.json:33]: why is this here?

File: tools/migrator/src/migrations/consolidate-props/transform.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      value2: 'value3',
+    },
+    fromPropType = 'string',
+    componentName = 'MyComponent',
+  } = options;
+
+  // Check if the element is on the page
+  const elements = source.findJSXElements(componentName);
+
+  // If there are not elements then return the original file
```
Comment (shaneeza) [inline tools/migrator/src/migrations/consolidate-props/transform.ts:32]: if there are no*

File: tools/migrator/src/utils/transformations/replaceJSXAttributes.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      .find(j.StringLiteral)
+      .forEach(literal => {
+        const isStringLiteral = typeof newPropValue === 'string';
+
+        if (isStringLiteral) {
+          literal.node.value = newPropValue;
+          return;
+        }
+
+        const value = literal.node.value;
```
Comment (shaneeza) [inline tools/migrator/src/utils/transformations/replaceJSXAttributes.ts:67]: Rename to `currentPropValue`

File: tools/build/config/rollup.config.mjs [hunk 11]
```diff
@@ -116,6 +116,8 @@ const external = [
   /^highlight/,
   /^lodash\//,
   /^react/,
+  'jscodeshift/src/Runner',
+  'is-git-clean',
```
Comment (shaneeza) [inline tools/build/config/rollup.config.mjs:120]: I'm not sure that I need this here.

> Definition: externals in Rollup refer to modules that are treated as external dependencies. These modules are not included in the bundle but are expected to be available in the environment where the bundle is executed.
> 
> Usage: You specify externals in your Rollup configuration to prevent certain modules from being bundled. This is particularly useful when you want to avoid duplicating dependencies that are already present in the target environment (e.g., a browser, Node.js, or another bundler).

File: tools/migrator/src/utils/jsx/insertJSXComment/transform.ts [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+import { insertJSXComment } from './insertJSXComment';
+
+export default function transformer(file: FileInfo, { jscodeshift: j }: API) {
+  const source = j(file.source);
+
+  source.findJSXElements('MyComponent').forEach(element => {
+    return getJSXAttributes(j, element, 'prop').forEach(el => {
+      // @ts-expect-error value does exist
+      const position = el.value.value?.value;
+      insertJSXComment(j, element, 'testing comment', position);
```
Comment (shaneeza) [inline tools/migrator/src/utils/jsx/insertJSXComment/transform.ts:16]: remove hardcoded comment

File: tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+      .forEach(literal => {
+        literal.node.value = newValueMapping;
+        removePropToRemove();
+        return;
+      });
+  }
+}
+
+/**
+ *
```
Comment (shaneeza) [inline tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts:133]: add comments

File: tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+  // if fromProp value is not in the mapping then remove that item from the atributes and return
+  if (!newValueMapping) {
+    removePropToRemove();
+    return;
+  }
+
+  // if the propToUpdate does not exist and there is a spread operator then return early since we don't know if the propToUpdate could be inside the spread
+  if (!toProp && hasSpreadOperator) {
+    // TODO: make string a constant
```
Comment (shaneeza) [inline tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts:97]: Make constant

File: tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts [hunk 15]
```diff
@@ -0,0 +1,164 @@
+// TODO: figure out correct types
+import type { ASTNode, ASTPath, JSXAttribute, Options } from 'jscodeshift';
+import type core from 'jscodeshift';
+
+import { insertJSXComment } from '../jsx/insertJSXComment/insertJSXComment';
+
+export interface ConsolidateJSXAttributesOptions extends Options {
+  j: core.JSCodeshift;
```
Comment (shaneeza) [inline tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts:8]: add docs

File: tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+    (attribute: ASTNode) => attribute.type !== 'JSXAttribute',
+  );
+
+  // removes the spread operator from the rest of the attributes
+  const allAttributesWithoutSpread = allAttributes.filter(
+    (attribute: ASTNode) => attribute.type === 'JSXAttribute',
+  );
+
+  // Finds the propToRemove from the list of attributes
+  const fromProp: ASTPath<JSXAttribute> = allAttributesWithoutSpread.find(
```
Comment (shaneeza) [inline tools/migrator/src/utils/transformations/consolidateJSXAttributes.ts:67]: this var name should be more clear. same for the other vars in here

File: tools/build/config/rollup.config.mjs (hunk 11 above)
Comment (TheSonOfThomp) [reply tools/build/config/rollup.config.mjs:120]: I don't think these should be here. Otherwise, iirc, this would require consuming apps to install `jscodeshift` as a dependency

File: tools/cli/package.json [hunk 17]
```diff
@@ -25,6 +25,7 @@
     "@lg-tools/test": "0.4.6",
     "@lg-tools/update": "0.1.8",
     "@lg-tools/validate": "0.1.10",
+    "@lg-tools/migrator": "0.0.1",
```
Comment (TheSonOfThomp) [inline tools/cli/package.json:28]: Should this be called `migrator`?  Would this be confused with the MongoDB Migrator product? Would `codemods` be a better package name?

File: tools/migrator/src/index.ts [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+
+    if (filepaths.length === 0) {
+      throw new Error(`No files found for ${files}`);
+    }
+
+    console.log(chalk.greenBright('filepaths:'), filepaths);
+    console.log(chalk.greenBright('Running migration:'), migration);
+
+    await jscodeshift.run(migrationFile, filepaths, {
+      ignorePattern: ['**/node_modules/**', '**/.next/**', '**/build/**'],
```
[the remark this answers is not in the document]
Comment (TheSonOfThomp) [reply tools/migrator/src/index.ts:57]: `/dist`? Should we also add an option for this if consumers want to customize this?

File: tools/migrator/README.md [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+npm install @lg-tools/migrator
+```
+
+## Usage
+
+`yarn lg migrator <migration> <path>`
+
+## Migrations
+
+_NOTE:_ These migrations are for testing purposes only
```
[the remark this answers is not in the document]
Comment (TheSonOfThomp) [reply tools/migrator/README.md:25]: Maybe add a contribution guide?

File: tools/migrator/src/migrations/consolidate-props/testing.tsx [hunk 20]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
```
Comment (TheSonOfThomp) [inline tools/migrator/src/migrations/consolidate-props/testing.tsx:1]: I'm a bit confused what this file is for

File: tools/migrator/src/migrations/consolidate-props/transform.ts [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+  const source = j(file.source);
+
+  const {
+    propToRemove = 'propToRemove',
+    propToUpdate = 'propToUpdate',
+    propMapping = {
+      value2: 'value3',
+    },
+    propToRemoveType = 'string',
+    componentName = 'MyComponent',
```
Comment (TheSonOfThomp) [inline tools/migrator/src/migrations/consolidate-props/transform.ts:34]: If I'm understanding correctly, these `migrations` that you've defined here are base transforms that we'll use for more specific mods?

File: tools/migrator/src/utils/checkGitStatus.ts [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+        chalk.greenBright('🥬 Thank you for using @lg-tools/migrator!'),
+      );
+      console.log(
+        chalk.red(
+          '\nBut before we continue, please stash or commit your git changes.',
+        ),
+      );
+      console.log(
+        chalk.yellow(
+          '\nYou may use the --force flag to override this safety check.',
```
Comment (TheSonOfThomp) [inline tools/migrator/src/utils/checkGitStatus.ts:38]: nit, you can add this as a single string with `\n` on L33

File: tools/migrator/src/utils/checkGitStatus.ts [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+    clean = isGitClean.sync(process.cwd());
+    errorMessage = 'Git directory is not clean';
+  } catch (err: any) {
+    if (err && err.stderr && err.stderr.indexOf('Not a git repository') >= 0) {
+      clean = true;
+    }
+  }
+
+  if (!clean) {
+    if (force) {
```
Comment (TheSonOfThomp) [inline tools/migrator/src/utils/checkGitStatus.ts:23]: should we check for `force` before the `try/catch`?

File: tools/migrator/src/migrations/consolidate-props/testing.tsx (hunk 20 above)
Comment (shaneeza) [reply tools/migrator/src/migrations/consolidate-props/testing.tsx:1]: For testing purposes I added this file to test that running `yarn lg migrator` works in the project. If you run `yarn lg migrator consolidate-props 'tools/migrator/src/migrations/consolidate-props/testing.tsx'`, it will update `testing.tsx`.

File: tools/migrator/src/migrations/consolidate-props/transform.ts (hunk 21 above)
Comment (shaneeza) [reply tools/migrator/src/migrations/consolidate-props/transform.ts:34]: correct!

File: tools/cli/package.json (hunk 17 above)
Comment (shaneeza) [reply tools/cli/package.json:28]: You might be right, `codemods` would be a better name to avoid confusion

File: tools/migrator/src/utils/checkGitStatus.ts (hunk 23 above)
Comment (shaneeza) [reply tools/migrator/src/utils/checkGitStatus.ts:23]: We can but then we won't know if the git directory is clean or not so we can't console log the explicit message, `Git directory is not clean`. 
Also, currently, if this isn't a git directory and the force flag is passed, we skip the force check, but if we check for force first we'll end up console logging `Forcibly continuing`

File: tools/migrator/src/index.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (shaneeza) [reply tools/migrator/src/index.ts]: Yeah i think we can do this, it can be an option that accepts an array of strings

File: tools/migrator/src/index.ts [hunk 24]
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-console */
+import chalk from 'chalk';
+import * as jscodeshift from 'jscodeshift/src/Runner';
+// import fs from 'fs';
+const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (shaneeza) [reply tools/migrator/src/index.ts:5]: I was struggling with this, import was not working and I'm not sure why

File: tools/cli/package.json (hunk 17 above)
Comment (shaneeza) [reply tools/cli/package.json:28]: I'll bring this up during dev sync!

File: tools/cli/src/index.ts [hunk 25]
```diff
@@ -166,6 +167,32 @@ cli
   )
   .action(validate);
 
+/** Migrator */
+cli
+  .command('migrator')
```
[the remark this answers is not in the document]
Comment (shaneeza) [reply tools/cli/src/index.ts:172]: If we rename the package to `codemod`, would the command also be `codemod`?

File: tools/cli/package.json (hunk 17 above)
Comment (shaneeza) [reply tools/cli/package.json:28]: `lg-tools/codemods`
`yarn lg migrate`
`yarn lg codemode`

File: .github/workflows/sizeDiff.yml [hunk 26]
```diff
@@ -19,5 +19,5 @@ jobs:
       - uses: preactjs/compressed-size-action@v2
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-          build-script: 'build'
+          build-script: 'preactBuild'
```
Comment (shaneeza) [inline .github/workflows/sizeDiff.yml:22]: wip

File: .changeset/great-cougars-explode.md [hunk 27]
```diff
@@ -0,0 +1,5 @@
+---
+'@lg-tools/cli': patch
+---
+
+Pre-release of cli.
```
Comment (TheSonOfThomp) [inline .changeset/great-cougars-explode.md:5]: Technically "Adds pre-release of migrator/codemods"

File: .github/workflows/sizeDiff.yml (hunk 26 above)
Comment (TheSonOfThomp) [reply .github/workflows/sizeDiff.yml:22]: should rm?

File: .changeset/short-pandas-obey.md [hunk 28]
```diff
@@ -0,0 +1,5 @@
+---
+'@lg-tools/migrator': patch
```
Comment (TheSonOfThomp) [inline .changeset/short-pandas-obey.md:2]: codemods now

File: tools/cli/src/index.ts [hunk 29]
```diff
@@ -166,6 +167,32 @@ cli
...
+  .description('Runs codemod transformations to upgrade LG components')
+  .argument(
+    '<migration>',
+    'One of the migrations from: https://github.com/mongodb/leafygreen-ui/blob/main/tools/migrator/README.md#migrations',
+  )
+  .argument(
+    '<path>',
+    'Files or directory to transform. Can be a glob like like src/**.test.js',
```
[the remark this answers is not in the document]
Comment (TheSonOfThomp) [reply tools/cli/src/index.ts:180]: I feel like `proccess.cwd` would be a safe default (the path that the command is run from)

File: tools/cli/src/index.ts [hunk 30]
```diff
@@ -169,28 +169,32 @@ cli
...
   )
   .argument(
     '<path>',
     'Files or directory to transform. Can be a glob like like src/**.test.js',
   )
-  .option('--dry', 'dry run (no changes are made to files)', false)
   .option(
-    '--print',
+    '--i, --ignore <items...>',
+    'Glob patterns to ignore. E.g. --i **/node_modules/** **/.next/**',
```
Comment (TheSonOfThomp) [inline tools/cli/src/index.ts:184]: technically `--ignore="**/node_modules/** **/.next/**"` (with `=` and quotes)

File: tools/codemods/CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,150 @@
...
+
+```
+src
+ ┣ migrations                                     # directory for all migrations
+ ┃ ┣ <name-of-migration>                          # directory for individual migration
+ ┃ ┃ ┣ tests                                      # directory for migration tests
+ ┃ ┃ ┃ ┗ <name-of-migration>                      # directory for a specific test
+ ┃ ┃ ┃ ┃ ┣ <name-of-migration>.input.tsx          # input file for test
+ ┃ ┃ ┃ ┃ ┣ <name-of-migration>.output.tsx         # output file for test
+ ┃ ┃ ┃ ┃ ┗ transform.spec.ts                      # jest test file
```
Comment (TheSonOfThomp) [inline tools/codemods/CONTRIBUTING.md:15]: are there too many nesting levels here?

File: tools/codemods/CONTRIBUTING.md [hunk 32]
```diff
@@ -0,0 +1,150 @@
...
+ ┃ ┣ transformations                              # directory for reusable transformations
+ ┃ ┃ ┣ <name-of-transformation>                   # directory for individual transformation
+ ┃ ┃ ┃ ┣ tests                                    # directory for transformation tests
+ ┃ ┃ ┃ ┃ ┣ <test-name>                            # directory for a specific test
+ ┃ ┃ ┃ ┃ ┃ ┣ <test-name>.input.tsx                # input file for test
+ ┃ ┃ ┃ ┃ ┃ ┣ <test-name>.output.tsx               # output file for test
+ ┃ ┃ ┃ ┃ ┃ ┗ transform.spec.ts
+ ┃ ┃ ┃ ┃ ┗ ...
+ ┃ ┃ ┃ ┣ <name-of-transformation>.ts              # transformation code
+ ┃ ┃ ┃ ┣ index.ts                                 # file to add exports
```
Comment (TheSonOfThomp) [inline tools/codemods/CONTRIBUTING.md:36]: same here

File: tools/codemods/src/index.ts [hunk 33]
```diff
@@ -12,35 +12,36 @@ export interface MigrateOptions {
...
+  ignore?: Array<string>;
 }
 
 export const migrator = async (
   migration: string,
   files: string | Array<string>,
   options: MigrateOptions = {},
 ) => {
-  // Gets the path of the migrations e.g: /Users/.../leafygreen-ui/tools/migrator/dist/migrations/[migration]/transform.js
+  console.log('😈', { options });
```
Comment (TheSonOfThomp) [inline tools/codemods/src/index.ts:23]: rm

File: tools/codemods/CONTRIBUTING.md [hunk 34]
```diff
@@ -0,0 +1,150 @@
+# Contributing to codemods
+
+## Getting Started
+
+All codemods/migrations can be found under `src/migrations` and each migration should have its own directory.
+
+```
+src
+ ┣ migrations                                     # directory for all migrations
```
Comment (shaneeza) [inline tools/codemods/CONTRIBUTING.md:9]: Maybe this should also be `codemods`

File: tools/migrator/src/index.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (TheSonOfThomp) [reply tools/migrator/src/index.ts]: Ah, there's something in our build system that doesn't like some built-in Node packages. Might need to use `fs-extra` (see some of the other tools)

File: tools/migrator/src/migrations/consolidate-props/testing.tsx (hunk 20 above)
Comment (TheSonOfThomp) [reply tools/migrator/src/migrations/consolidate-props/testing.tsx]: What's the difference between `/testing.tsx` and the`/*.input.tsx`/`/*.output.tsx` files?

Comment (shaneeza) [reply tools/migrator/src/migrations/consolidate-props/testing.tsx]: `/testing.tsx` is tested by running `yarn lg migrator consolidate-props` and `/*.input.tsx/`/`*.output.tsx` is tested with jest using,  `yarn test /tools/codemods`

File: tools/codemods/CONTRIBUTING.md (hunk 31 above)
Comment (shaneeza) [reply tools/codemods/CONTRIBUTING.md:15]: i think that i can get rid of the directories for a specific test, Ln12

File: tools/cli/src/index.ts (hunk 30 above)
Comment (shaneeza) [reply tools/cli/src/index.ts:184]: That doesn't return the correct array of options
```
--i="one two"
```
```
ignore: [ 'one two' ],
```
but this does
```
--i one two
```
```
ignore: [ 'one', 'two' ],
```
it might be because I'm using `option` and not `argument`?

File: tools/migrator/src/migrations/consolidate-props/testing.tsx (hunk 20 above)
Comment (TheSonOfThomp) [reply tools/migrator/src/migrations/consolidate-props/testing.tsx]: I'm not sure these files should stay then, since they're only for manual testing. The only files used in automated tests are the `/*.input.tsx/`/`*.output.tsx` files

File: tools/migrator/src/index.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (TheSonOfThomp) [reply tools/migrator/src/index.ts]: Thinking on this more, we should use `fs-extra` for consistency across platforms

File: tools/codemods/rollup.config.mjs [hunk 35]
```diff
@@ -0,0 +1,48 @@
...
+  //   // This creates the dist/cjs dir. It includes the /codemods dir which includes .js files
+  //   output: {
+  //     ...esmConfig.output,
+  //     format: 'cjs', // overrides esm format from esmConfig.output
+  //     entryFileNames: '[name].js',
+  //     preserveModules: true,
+  //     exports: 'auto',
+  //     dir: 'dist/cjs/',
+  //   },
+  // },
```
Comment (TheSonOfThomp) [inline tools/codemods/rollup.config.mjs:47]: rm

File: tools/cli/src/index.ts (hunk 30 above)
Comment (TheSonOfThomp) [reply tools/cli/src/index.ts:184]: Huh, i was wrong then!

File: tools/migrator/src/migrations/consolidate-props/testing.tsx (hunk 20 above)
Comment (shaneeza) [reply tools/migrator/src/migrations/consolidate-props/testing.tsx]: i'll remove them

**Review by TheSonOfThomp — APPROVED** [review]
Awesome! LGTM!
